cogs/subjects.py

```
@@ -0,0 +1,272 @@
+        roles: list[Role] = member.roles
+
+        subjects: list[Role] = await self.get_possible_subjects(ctx, member, roles)
+
+        if not subjects:
+            return
+
+        if subject not in [subject.name for subject in subjects]:
+            embed = Embed(title="No permission",
+                          description=f"You can't be assigned to this subject.\n"
+                                      f"Please contact an admin if you think this is a mistake")
+            await ctx.reply(content=f"<@{member.id}>", embed=embed)
+            return
+
+        if subject not in [subject.name for subject in subjects if subject not in roles]:
+            embed = Embed(title="Already assigned",
+                          description=f"You are already assigned to this subject")
+            await ctx.reply(content=f"<@{member.id}>", embed=embed)
+            return
+
+        role: Role = [role for role in subjects if role.name == subject][0]
+        await member.add_roles(role)
+        embed = Embed(title="successfully assigned",
+                      description=f"Assigned you to <@&{role.id}>")
+        await ctx.reply(content=f"<@{member.id}>", embed=embed)
+        return
+
+    @subject.command(pass_context=True,
+                     aliases=["rem", "rm"],
+                     name="remove")
+    async def subject_remove(self, ctx: Context, subject: str):
+        """
+        opts-out a user into the specified subject
+
+        Args:
+            ctx: The command context provided by the discord.py wrapper.
+
+            subject: The Subject to opt into
+        """
+        member: Union[Member, User] = ctx.author
+        roles: list[Role] = member.roles
+
+        subjects: list[Role] = await self.get_possible_subjects(ctx, member, roles)
+
+        if not subjects:
+            return
+
+        if subject not in [subject.name for subject in subjects if subject in roles]:
+            embed = Embed(title="Not assigned",
```

This should throw an error so that it can be treated as an error.